lib/writers/s3_test.go

```diff
@@ -0,0 +1,118 @@
+package writers_test
+
+import (
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	platformv1alpha1 "github.com/syntasso/kratix/api/v1alpha1"
+	"github.com/syntasso/kratix/lib/writers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var _ = Describe("S3", func() {
+	Describe("NewS3Writer", func() {
+		var (
+			logger         logr.Logger
+			dest           platformv1alpha1.Destination
+			stateStoreSpec platformv1alpha1.BucketStateStoreSpec
+		)
+
+		BeforeEach(func() {
+			logger = ctrl.Log.WithName("setup")
+			stateStoreSpec = platformv1alpha1.BucketStateStoreSpec{
+				Endpoint:   "example.com",
+				Insecure:   true,
+				AuthMethod: "secretAccessKey",
+			}
+
+			dest = platformv1alpha1.Destination{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "test",
+				},
+				Spec: platformv1alpha1.DestinationSpec{},
+			}
+
+		})
+
+		Context("secretAccessKey", func() {
```

Comment:
nit: is this the context for the `authMethod`? should it be `accessKey` instead of `secretAccessKey`?

Reply:
oops yep. Originally I called the auth method `secretAccessKey`, decided to rename to `accessKey` but missed this! thanks